### PR TITLE
Remove unused exceptions

### PR DESCRIPTION
**Pull request by mea36**
Fixes https://github.com/Expensify/Expensify/issues/65018 and https://github.com/Expensify/Expensify/issues/67763

**Tests**
**Timeout exception**
```
>>> try {$db->query('select 1 from table;');} catch (Exception $e) {}
>>> $e
=> Expensify\Bedrock\Exceptions\BedrockError {#403
     #message: "555 Timeout peeking command",
     #code: 555,
     #line: 71,
   }
```
**Query failed exception**
```
>>> try {$db->query('select;');} catch (Exception $e) {}
>>> $e
=> Expensify\Bedrock\Exceptions\BedrockError {#404
     #message: "502 Query failed - near ";": syntax error",
     #code: 502,
     #line: 67,
   }
```

### CODE REVIEW

File: src/DB.php [hunk 1]
```diff
@@ -66,11 +64,11 @@ public function run(string $sql, bool $idempotent, int $timeout = 60000000): Res
         ));
 
         if ($response->getCode() === self::CODE_QUERY_FAILED) {
```
Comment (mea36) [inline src/DB.php]: I don't know if we want to keep this here anyway.  I'm assume we want to remove it but asking nonetheless

Comment (iwiznia) [reply src/DB.php]: I think we can remove it, the error would say query failed anyway, right?

**Comment by mea36** [comment]
@iwiznia updated
I removed `CODE_QUERY_FAILED` and updated the exception to have a better message and include the response code

I'll update the version after you approve this

**Comment by mea36** [comment]
Also, I found a bug, check it out https://github.com/Expensify/Expensify/issues/67763

**Comment by iwiznia** [comment]
As a general rule, please add hold or wip to the name of PRs in this repo, since we need to create a new version and we don't want to merge it by mistake before we create the version.

File: src/DB.php (hunk 1 above)
Comment (mea36) [reply src/DB.php]: I added this back so we can have an exception message that includes the error

File: src/Client.php [hunk 2]
```diff
@@ -601,7 +601,7 @@ private function extractResponseHeaders(array $responseHeaderLines)
         $responseHeaders = [];
         foreach ($responseHeaderLines as $responseHeaderLine) {
             // Try to split this line
-            $nameValue = explode(":", $responseHeaderLine);
+            $nameValue = explode(":", $responseHeaderLine, 2);
```
Comment (quinthar) [inline src/Client.php]: Ooh, nice catch!

File: src/DB.php (hunk 1 above)
Comment (quinthar) [reply src/DB.php]: I don't follow this -- why don't we call `$response->getError()` on all non-200 responses?  Why is 502 special?

**Comment by mea36** [comment]
Updated and added tests.  I'll add the version when this is approved

File: src/DB.php (hunk 1 above)
Comment (mea36) [reply src/DB.php]: `error` is only added when we throw a 502

https://github.com/Expensify/Bedrock/blob/ce238f0/plugins/DB.cpp#L78-L83

Comment (quinthar) [reply src/DB.php]: Ah, makes sense.  thanks!

File: src/Client.php [hunk 3]
```diff
@@ -10,6 +10,8 @@
 use Psr\Log\LoggerInterface;
 use Psr\Log\NullLogger;
 
+define('TRAVIS_RUNNING', isset($_SERVER['TRAVIS_COMMIT']));
```
Comment (iwiznia) [inline src/Client.php]: Why did you add this? I don't think we can do this, at least you need to check that the constant is not defined before.

Comment (mea36) [reply src/Client.php]: Sorry for the delay, this fell off of my radar.
I did this because phan was yelling at me https://travis-ci.org/Expensify/Bedrock-PHP/builds/313619228

But you're right, I should check if this is defined first

Comment (mea36) [reply src/Client.php]: Actually, after taking a 2nd look at the code, I think I should add a comment for phan to ignore the error
```
FAIL! src/Client.php: src/Client.php:436 PhanUndeclaredConstant Reference to undeclared constant \TRAVIS_RUNNING
FAIL! src/Client.php: src/Client.php:640 PhanUndeclaredConstant Reference to undeclared constant \TRAVIS_RUNNING
```

**Comment by mea36** [comment]
@iwiznia updated.  I'll add a version when you approve these changes

**Comment by mea36** [comment]
@iwiznia updated
